source/grid/grid_tools.cc

```diff
@@ -5882,67 +5887,104 @@ namespace GridTools
+          }
+
+        // helper function to determine if a bounding box is valid
+        const auto is_valid = [](const auto &bb) {
+          for (unsigned int i = 0; i < spacedim; ++i)
+            if (bb.get_boundary_points().first[i] >
+                bb.get_boundary_points().second[i])
+              return false;
 
-          // linearize vector of vectors
-          std::vector<std::pair<BoundingBox<spacedim>, unsigned int>>
-            boxes_and_ranks;
+          return true;
+        };
 
-          for (unsigned rank = 0; rank < global_bboxes_to_be_used->size();
-               ++rank)
-            for (const auto &box : (*global_bboxes_to_be_used)[rank])
-              if (is_valid(box))
-                boxes_and_ranks.emplace_back(box, rank);
+        // linearize vector of vectors
+        std::vector<std::pair<BoundingBox<spacedim>, unsigned int>>
+          boxes_and_ranks;
 
-          // pack boxes into r-tree
-          const auto tree = pack_rtree(boxes_and_ranks);
+        for (unsigned rank = 0; rank < global_bboxes_to_be_used->size(); ++rank)
+          for (const auto &box : (*global_bboxes_to_be_used)[rank])
+            if (is_valid(box))
+              boxes_and_ranks.emplace_back(box, rank);
 
-          // loop over all entities
-          for (unsigned int i = 0; i < entities.size(); ++i)
-            {
-              // create a bounding box with tolerance
-              const auto bb =
-                BoundingBox<spacedim>(entities[i]).create_extended(tolerance);
+        // pack boxes into r-tree
+        const auto tree = pack_rtree(boxes_and_ranks);
 
-              // determine ranks potentially owning point/bounding box
-              std::set<unsigned int> my_ranks;
+        // loop over all entities
+        for (unsigned int i = 0; i < entities.size(); ++i)
+          {
+            // create a bounding box with tolerance
+            const auto bb =
+              BoundingBox<spacedim>(entities[i]).create_extended(tolerance);
+
+            // determine ranks potentially owning point/bounding box
+            std::set<unsigned int> my_ranks;
 
-              for (const auto &box_and_rank :
-                   tree | boost::geometry::index::adaptors::queried(
-                            boost::geometry::index::intersects(bb)))
-                my_ranks.insert(box_and_rank.second);
+            for (const auto &box_and_rank :
+                 tree | boost::geometry::index::adaptors::queried(
+                          boost::geometry::index::intersects(bb)))
+              my_ranks.insert(box_and_rank.second);
+
+            for (const auto rank : my_ranks)
+              ranks_and_indices.emplace_back(rank, i);
+          }
+      };
 
-              for (const auto rank : my_ranks)
-                ranks_and_indices.emplace_back(rank, i);
+#if defined(DEAL_II_WITH_ARBORX)
+      if (global_bboxes.size() == 1)
+        {
+          if constexpr (spacedim != 1)
```

Comment:
I'm confused shouldn't that be `3`? Also, we shouldn't error out when the `spacedim` requirement is not met but rather copy to 3d data structures for `ArborX`.

Reply:
Ah, I was reading this as `spacedim==1`. My bad.

Reply:
Yes, otherwise one gets https://github.com/dealii/dealii/pull/15538#issuecomment-1624128212

Frage. The situation may not be the best right now because the current implementation throws if ArborX is available and spacedim == 1, while without ArborX it doesn't.

Reply:
I would include that in your `USE_ARBORX` macro (turning it into a `constexpr` variable) and deal with `spacedim==1` support in `ArborX` in a follow-up.

Reply:
So done, thanks.